src/NuGet.Clients/NuGet.PackageManagement.VisualStudio/IDE/DeferredProjectWorkspaceService.cs

```diff
@@ -59,51 +60,29 @@ internal sealed class DeferredProjectWorkspaceService : IDeferredProjectWorkspac
             return fileReferenceResult.Select(f => workspace.MakeRooted(f.Path));
         }
 
-        public async Task<IMSBuildProjectDataService> GetMSBuildProjectDataService(string projectFilePath, string targetFramework = "")
+        public async Task<IMSBuildProjectDataService> GetMSBuildProjectDataServiceAsync(
+            string projectFilePath, string targetFramework)
         {
-            return await NuGetUIThreadHelper.JoinableTaskFactory.RunAsync(async () =>
-            {
-                await NuGetUIThreadHelper.JoinableTaskFactory.SwitchToMainThreadAsync();
-                var factory = SolutionWorkspaceService.GetService(typeof(IVsSolutionMSBuildProjectServiceFactory)) as IVsSolutionMSBuildProjectServiceFactory;
+            Assumes.NotNullOrEmpty(projectFilePath);
 
-                if (factory == null)
-                {
-                    throw new ArgumentNullException(nameof(factory));
-                }
+            await NuGetUIThreadHelper.JoinableTaskFactory.SwitchToMainThreadAsync();
 
-                if (string.IsNullOrEmpty(targetFramework))
-                {
-                    return await factory.GetMSBuildProjectDataServiceAsync(projectFilePath);
-                }
-                else
-                {
-                    var projectProperties = new Dictionary<string, string>
-                    {
-                        { "TargetFramework", targetFramework }
-                    };
-                    return await factory.GetMSBuildProjectDataServiceAsync(projectFilePath, projectProperties: projectProperties);
-                }
-            });
-        }
+            var factory = SolutionWorkspaceService.GetService<IVsSolutionMSBuildProjectServiceFactory>();
+            Assumes.Present(factory);
 
-        public async Task<IEnumerable<MSBuildProjectItemData>> GetProjectItemsAsync(IMSBuildProjectDataService dataService, string itemType)
-        {
-            if (dataService == null)
+            if (string.IsNullOrEmpty(targetFramework))
             {
-                throw new ArgumentNullException(nameof(dataService));
+                return await factory.GetMSBuildProjectDataServiceAsync(projectFilePath);
             }
-
-            return await dataService.GetProjectItems(itemType);
-        }
-
-        public async Task<string> GetProjectPropertyAsync(IMSBuildProjectDataService dataService, string propertyName)
-        {
-            if (dataService == null)
+            else
             {
-                throw new ArgumentNullException(nameof(dataService));
+                var projectProperties = new Dictionary<string, string>
+                    {
+                        { "TargetFramework", targetFramework }
+                    };
+                return await factory.GetMSBuildProjectDataServiceAsync(
+                    projectFilePath, projectProperties: projectProperties);
```

Comment:
The factory method has more parameters with default values. This statement selectively overrides one of them.